addons/pos_loyalty/static/src/js/TicketScreen.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module **/
+
+import TicketScreen from 'point_of_sale.TicketScreen';
+import Registries from 'point_of_sale.Registries';
+
+/**
+ * This override is introduced so that when accidentally refunding ewallet reward
+ * ewallet balance is still properly updated.
+ * This is accomplished by replacing the reward product with the top up product
+ * of the ewallet program.
+ * This can be extended for gift_card programs but it's not implemented yet.
+ * @param {owl.Component} TicketScreen
+ */
+export const PosLoyaltyTicketScreen = (TicketScreen) =>
+    class PosLoyaltyTicketScreen extends TicketScreen {
+        _getToRefundDetail(orderline) {
+            const toRefundDetail = super._getToRefundDetail(orderline);
+            // Intercept the result. Replace with top-up products if refunding ewallet reward lines.
```

Comment:
We shouldn't allow the refund of an e-wallet product, this will open to too many strange scenarios. The ewallet isn't acting like a bank account where we can transfer money in and **out** of it. 
At the moment, by testing your PR, it is possible to buy an ewallet, then sell it back and the amount of the ewallet didn't change.

Reply:
And i'm talking about the ewallet top-up product and also the one used for the reward line

Reply:
> At the moment, by testing your PR, it is possible to buy an ewallet, then sell it back and the amount of the ewallet didn't change.

Can you show me that part that doesn't work?

Reply:
> The ewallet isn't acting like a bank account where we can transfer money in and out of it.

I think you got this wrong. I know it's not a bank. But it should still be allow to have "in" and "out" transactions. Besides, ewallet is the best place to put refunded products.

Reply:
You're right. I should just prevent refunding ewallet lines. So many things to consider in order to make it properly work.

Reply:
When I say "out", it's more about the fact that credit inside the wallet cannot be converted back to real money. A good example of this is the ewallet from Steam where you can put money in it in order to buy games but it's not possible to get the money back out of it. I guess this is more a spec that needs to be further discussed rather than a technical review
Here's where you can refund the top-up ewallet to get back your money and where you still have money in it (in my eyes, this shoud be forbidden): https://watch.screencastify.com/v/hyiUt3EAewxb33qCVymD
I was wrong (or maybe you changed it after we noticed it a while ago), the refund of the ewallet reward line is working fine